src/test/java/uk/gov/hmcts/reform/bulkscanprocessor/validation/OcrValidatorTest.java

```diff
@@ -456,6 +470,104 @@ public void should_log_info_when_ocr_is_present_but_theres_not_service_configure
+            .willThrow(HttpServerErrorException.create(
+                HttpStatus.INTERNAL_SERVER_ERROR,
+                "internal server error message",
+                HttpHeaders.EMPTY,
+                null,
+                null
+            ));
+        given(ocrValidationRetryManager
+                  .setRetryDelayIfPossible(
+                      any(BlobClient.class),
+                      anyString()
+                  ))
+            .willReturn(true);
+
+        // when
+        OcrValidationServerSideException exception = catchThrowableOfType(
+            () -> ocrValidator.assertOcrDataIsValid(envelope, blobClient, LEASE_ID),
+            OcrValidationServerSideException.class
+        );
+
+        // then
+        assertThat(exception.getMessage()).matches(
+            "Error calling validation endpoint. Url: https://example.com/validate-ocr, "
+                + "DCN: (.*), doc type: Form, doc subtype: x, "
+                + "envelope: file.zip.");
+    }
+
+    @Test
+    void should_throw_exception_if_service_responded_with_500_and_retry_delay_is_not_possible() {
+        // given
+        InputScannableItem scannableItemWithOcr = doc(FORM, "form", sampleOcr());
+        InputEnvelope envelope = envelope(
+            PO_BOX,
+            asList(
+                scannableItemWithOcr,
+                doc(OTHER, "other", null)
+            ),
+            SUPPLEMENTARY_EVIDENCE
+        );
+
+        given(presenceValidator.assertHasProperlySetOcr(envelope.scannableItems))
+            .willReturn(Optional.of(scannableItemWithOcr));
+
+        given(containerMappings.getMappings())
+            .willReturn(singletonList(
+                new Mapping("c", "j", envelope.poBox, VALIDATION_URL, true, true)
+            ));
+
+        given(authTokenGenerator.generate()).willReturn(S2S_TOKEN);
+
+        given(client.validate(anyString(), any(FormData.class), anyString(), anyString()))
+            .willThrow(HttpServerErrorException.create(
+                HttpStatus.INTERNAL_SERVER_ERROR,
+                "internal server error message",
+                HttpHeaders.EMPTY,
+                null,
+                null
+            ));
+        given(ocrValidationRetryManager
+                  .setRetryDelayIfPossible(
+                      any(BlobClient.class),
+                      anyString()
+                  ))
+            .willReturn(false);
+
+        // when
+        Optional<OcrValidationWarnings> warnings = ocrValidator.assertOcrDataIsValid(envelope, blobClient, LEASE_ID);
+
+        // then
+        assertThat(warnings).isPresent();
+        assertThat(warnings.get().documentControlNumber).isEqualTo(scannableItemWithOcr.documentControlNumber);
+        assertThat(warnings.get().warnings).containsExactly("OCR validation was not performed due to errors");
+        verify(client).validate(anyString(), any(FormData.class), anyString(), anyString());
```

Comment:
test says `should_throw_exception_if....`. Looks like it does not throw an exception.

Reply:
Renamed to `should_raise_ocr_validation_warnings_if_service_responded_with_500_and_retry_delay_is_not_possible`.